### PR TITLE
PF-75: API REST Info about user

### PR DESCRIPTION
**Pull request by tpambor**
## Release Notes

- **New Features**
  - Introduced a new user management blueprint (`BlueprintUser`) for enhanced user-related functionalities.
  - Added an API endpoint to retrieve user information with token validation.
  
- **Bug Fixes**
  - Improved error handling for user retrieval scenarios, ensuring appropriate responses for missing users and invalid tokens.

- **Tests**
  - Added comprehensive unit tests for user-related API endpoints, enhancing test coverage for user retrieval and token validation scenarios.

- **Chores**
  - Updated dependencies and configuration files to improve project structure and linting rules.

### CODE REVIEW

**Comment by tpambor** [comment]
@coderabbitai full review

File: blueprints/user.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+blp = Blueprint('Users', __name__)
+
+
+@class_route(blp, '/api/v1/users/me')
+class UserInfo(MethodView):
+    init_every_request = False
+
+    @requires_token
+    def get(self, token: dict[str, Any], user_repo: UserRepository = Provide[Container.user_repo]) -> Response:
```
[the remark this answers is not in the document]
Comment (tpambor) [reply blueprints/user.py:20]: They are guaranteed to exist in when used with `@requires_token`

File: blueprints/util.py [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
 
-from flask import Blueprint, Response
+from flask import Blueprint, Request, Response, request
 from flask.views import MethodView
 from marshmallow import ValidationError
+from tightwrap import wraps
+
+
+class APIGatewayRequest(Request):
+    user_token: dict[str, Any]
```
[the remark this answers is not in the document]
Comment (tpambor) [reply blueprints/util.py:12]: This is only for type checking purposes

File: blueprints/util.py [hunk 3]
```diff
@@ -29,3 +34,24 @@ def validation_error_response(err: ValidationError) -> Response:
         return error_response(msg, 400)
 
     raise NotImplementedError('Validation error response for non-dict messages not implemented.')  # pragma: no cover
+
+
+def requires_token(f: Callable[..., Response]) -> Callable[..., Response]:
+    @wraps(f)
+    def decorated_function(*args, **kwargs) -> Response:  # type: ignore[no-untyped-def] # noqa: ANN002, ANN003
+        if hasattr(request, 'user_token') and cast(APIGatewayRequest, request).user_token is not None:
+            req = cast(APIGatewayRequest, request)
```
[the remark this answers is not in the document]
Comment (tpambor) [reply blueprints/util.py:43]: It is checked with the if

File: blueprints/util.py [hunk 4]
```diff
@@ -1,10 +1,15 @@
 import json
 from collections.abc import Callable
-from typing import Any
+from typing import Any, cast
 
-from flask import Blueprint, Response
+from flask import Blueprint, Request, Response, request
 from flask.views import MethodView
 from marshmallow import ValidationError
+from tightwrap import wraps
```
[the remark this answers is not in the document]
Comment (tpambor) [reply blueprints/util.py:8]: tightwrap works better with typing

**Review by VivianaOj — APPROVED** [review]

**Pull request merged**